config.go

```diff
@@ -41,14 +41,26 @@ const (
 	_packageName       = "yarpc"
 )
 
+// LogLevelConfig configures the levels at which YARPC logs various things.
+type LogLevelConfig struct {
+	// Level at which application errors are logged.
+	//
+	// Defaults to ErrorLevel.
+	ApplicationError *zapcore.Level
```

Comment:
nit: Can we name this `ApplicationErrorLevel`?

Reply:
I'd opt for us not too as the naming becomes redundant when reading the struct and field name.
`LogLevelConfig.ApplicationError` vs `LogLevelConfig.ApplicationErrorLevel`. The latter sounds repetitive.

Reply:
I'm open to changing it.

I figured it shouldn't be repeated because that this is a level is abundantly
clear due to the following,

- We're in a struct named `*Levels`
- The type of the field is `*Level`

What do you think?

Reply:
I vote for no stutter in the naming.

Reply:
I'm persuaded. Carry on.